docs/developer/design/dashboard-architecture.md

```diff
@@ -0,0 +1,660 @@
+11. [Extension Points](#extension-points)
+12. [Development Principles and Technologies](#development-principles-and-technologies)
+
+## Overview
+
+The Beamlime dashboard is a real-time data visualization system that follows a layered architecture with clear separation of concerns between presentation, application logic, and infrastructure. The system is designed for live display of raw and processed detector data with configurable processing parameters, using dependency injection patterns for testability and maintainability.
+
+A key architectural principle is the separation between **Pydantic models** (used for Kafka message validation and backend communication) and **Param models** (used for GUI widgets and user interaction).
+
+The dashboard processes 1-D and 2-D data displayed using Holoviews with update rates on the order of 1Hz. Data updates are received via Kafka streams, and user controls result in configuration updates published to Kafka topics for backend consumption.
+
+## System Context: Dashboard and Kafka Integration
+
+The Beamlime dashboard operates within a Kafka-based system, interacting with multiple backend services via Kafka topics for both data and configuration.
+
+```mermaid
+flowchart TD
+    ConfigTopic["Beamlime Config Kafka Topic"]
+    DataTopic["Beamlime Data Kafka Topic"]
+
+    subgraph BackendServices
+        MonitorData["monitor_data service"]
+        DetectorData["detector_data service"]
+        DataReduction["data_reduction service"]
+    end
+
+    DashboardApp["Beamlime Dashboard"]
+
+    %% Data publishing
+    MonitorData -- Publishes --> DataTopic
+    DetectorData -- Publishes --> DataTopic
+    DataReduction -- Publishes --> DataTopic
+
+    %% Data consumption by dashboard
+    DataTopic -- Feeds --> DashboardApp
+
+    %% Config flows
+    DashboardApp -- Publishes config --> ConfigTopic
+    DashboardApp -- Consumes config --> ConfigTopic
+
+    ConfigTopic -- Consumed by --> MonitorData
+    ConfigTopic -- Consumed by --> DetectorData
+    ConfigTopic -- Consumed by --> DataReduction
+
+    DataReduction -- Publishes config (workflow specs) --> ConfigTopic
+```
+
+**Key Points:**
+- Backend Services publish data streams to a single Kafka topic
+- The Dashboard consumes this data topic, feeding into internal `DataService` components
+- The Dashboard both publishes to and consumes from the config topic
+- All backend services consume the config topic for configuration updates
+- The `data_reduction` service can publish configuration messages (workflow specs, status)
+
+## High-Level Architecture
+
+```mermaid
+graph TD
+    subgraph "External Systems"
+        K1[Kafka Data Streams]
+        K2[Kafka Config Topic]
+    end
+
+    subgraph "Infrastructure Layer"
+        KB[KafkaBridge]
+        MS[MessageSource]
+    end
+
+    subgraph "Application Layer"
+        CS[ConfigService]
+        DS[DataServices]
+        PM[Param Models]
+        WC[WorkflowController]
```

Comment:
I can't find a `WorkflowController` in the repo. Is this name correct?

Reply:
Anything related to workflows is in the follow-up branch, adding data reduction.